### PR TITLE
Remove global storage

### PR DESCRIPTION
**Pull request by henryju**

### CODE REVIEW

File: core/src/main/java/org/sonarsource/sonarlint/core/client/api/connected/ConnectedAnalysisConfiguration.java [hunk 1]
```diff
@@ -53,9 +51,7 @@ public String toString() {
   private String generateString() {
     var sb = new StringBuilder();
     sb.append("[\n");
-    if (projectKey != null) {
-      sb.append("  projectKey: ").append(projectKey).append("\n");
-    }
+    sb.append("  projectKey: ").append(projectKey).append("\n");
```
Comment (damien-urruty-sonarsource) [inline core/src/main/java/org/sonarsource/sonarlint/core/client/api/connected/ConnectedAnalysisConfiguration.java:54]: Nitpick: the last line could also be chained

File: server-api/src/main/java/org/sonarsource/sonarlint/core/serverapi/organization/OrganizationApi.java [hunk 2]
```diff
@@ -36,25 +35,11 @@ public OrganizationApi(ServerApiHelper helper) {
...
-  private List<ServerOrganization> listUserOrganizations(ServerVersionAndStatusChecker serverChecker, ProgressMonitor progress) {
-    checkServer(serverChecker, progress);
-    return fetchUserOrganizations(progress.subProgress(0.2f, 1.0f, "Fetch organizations"));
-  }
-
-  private static void checkServer(ServerVersionAndStatusChecker serverChecker, ProgressMonitor progress) {
-    progress.setProgressAndCheckCancel("Check server version", 0.1f);
-    serverChecker.checkVersionAndStatus();
-    progress.setProgressAndCheckCancel("Fetch organizations", 0.2f);
+    return fetchOrganization(organizationKey, progress);
```
Comment (damien-urruty-sonarsource) [inline server-api/src/main/java/org/sonarsource/sonarlint/core/serverapi/organization/OrganizationApi.java:42]: Nitpick: we can probably get rid of those 2 delegate methods (list and get)

**Review by damien-urruty-sonarsource — APPROVED** [review]
Nitpick comments else LGTM!

File: core/src/main/java/org/sonarsource/sonarlint/core/client/api/connected/ConnectedAnalysisConfiguration.java (hunk 1 above)
Comment (henryju) [reply core/src/main/java/org/sonarsource/sonarlint/core/client/api/connected/ConnectedAnalysisConfiguration.java:54]: :+1:

File: server-api/src/main/java/org/sonarsource/sonarlint/core/serverapi/organization/OrganizationApi.java (hunk 2 above)
Comment (henryju) [reply server-api/src/main/java/org/sonarsource/sonarlint/core/serverapi/organization/OrganizationApi.java:42]: :+1:

**Pull request merged**